File: ruby/lib/runapi/suno/resources/add_samples.rb
module RunApi
  module Suno
    module Resources
      class AddSamples < AudioAction
        ENDPOINT = "/api/v1/suno/add_samples"
        ACTION = "add-samples"

        def create(options: nil, **params)
          if params[:end_seconds].to_f <= params[:start_seconds].to_f
            raise RunApi::Core::ValidationError, "end_seconds must be greater than start_seconds"
          end

          super
        end
      end
    end
  end
end

File: python/src/runapi/suno/resources/add_samples.py
from .audio_actions import AudioAction
from runapi.core.errors import ValidationError

class AddSamples(AudioAction):
    ENDPOINT = "/api/v1/suno/add_samples"
    ACTION = "add-samples"

    def create(self, options=None, **params):
        if params.get("end_seconds", 0) <= params.get("start_seconds", 0):
            raise ValidationError("end_seconds must be greater than start_seconds")
        return super().create(options=options, **params)
